### PR TITLE
[NEW] User Status Messages

### PR DESCRIPTION
**Pull request by wreiske**
This pull request adds the ability for users to set a custom status message by editing their profile or using a new `/status` slash command.

Status messages show up in direct messages on the header, as well as in the user info sidebar.

### UPDATE
This pull request has changed recently. Please see https://github.com/RocketChat/Rocket.Chat/pull/12710#issuecomment-473618723

### CODE REVIEW

**Comment by rodrigok** [comment]
@engelgabriel @thiagosanchz what you guys think about this?

**Comment by geekgonecrazy** [comment]
Nice!  Would be great to figure out how to surface this where it could be accessed quicker.  But I think this is great way to do it!  Especially with the slash command which will make it instantly usable even on mobile

**Comment by Hudell** [comment]
I had been working on #9353 to try and get it merged for a while, maybe both PRs could be merged together as both seem to have features the other lack.

**Comment by richardt3** [comment]
Looks good - it's something my users keep asking for. Obviously would like to see it in the same place as the current status change, but the slash command makes up for it. Can we get this merged to live pls!

**Comment by geekgonecrazy** [comment]
@Hudell how hard would this be?  Are both using close enough approches one could be merged and then the other merged to extend the functionality?

**Comment by wreiske** [comment]
I'm not sure. I haven't looked too far into the code for the other system. I think for the sake of getting this in, we might as well merge this (since it includes the slash command to allow mobile status changes). My team can commit to working on the extra elements needed to set status from within the UI, as well as working with the IOS and Android apps to get it implemented there.

**Comment by mickymiek** [comment]
What's required to get this in? We're very interested in this feature. We could provide help too if needed :)

**Comment by fifou17** [comment]
when will be merged guys ?? any plan ?

**Comment by wreiske** [comment]
This will need to be re-worked since a lot of the surrounding code has been moved around. 

I'm willing to put time towards this again, but only with @geekgonecrazy 's or others OK. I don't want to spend more time on a pull-request that is a no-go. Let me know!

**Comment by geekgonecrazy** [comment]
I personally still like this and would like to have it in.  Even if the UI changed a bit after implementation.

> @engelgabriel @thiagosanchz what you guys think about this?

@engelgabriel @thiagosanchz Ping! Still would like thoughts :)

**Comment by sampaiodiego** [comment]
a lot will need to be changed code wise though as I'm about to move everything away from meteor packages..

**Comment by wreiske** [comment]
> a lot will need to be changed code wise though as I'm about to move everything away from meteor packages..

Let me know when it's safe for me to start working on this again. I saw a commit recently that moved a bunch of stuff out of the meteor packages. I'll plan a weekend to hammer out the status messages if you think the current implementation will be okay as a starting point. 

**Essentially:**
- /status slash command to set status (requires no changes to mobile apps to immediately allow changing of statuses while they implement UI for it)
- Change status on the accounts page
- Display status in the user bar 
- Display status in the user info side panel

**Comment by Hudell** [comment]
I'll be working on merging this (along with the other status PR too) next week.

@wreiske if you want to solve the conflicts on this one, it would be a huge help!

**Comment by wreiske** [comment]
> I'll be working on merging this (along with the other status PR too) next week.
> 
> @wreiske if you want to solve the conflicts on this one, it would be a huge help!

I've updated to the latest build... Also added an Edit Status modal dialog. Trying to figure out how to push/update this pull request. I may need to just start a new pull-request.

user-status-messages mp4

**Comment by wreiske** [comment]
Updated to latest greatest!

**Comment by wreiske** [comment]
Added rocket.chat output message when using the status slash command. 

user-status-message-slash mp4

File: app/api/server/v1/users.js [hunk 1]
```diff
@@ -378,6 +380,7 @@ API.v1.addRoute('users.updateOwnBasicInfo', { authRequired: true }, {
 			email: this.bodyParams.data.email,
 			realname: this.bodyParams.data.name,
 			username: this.bodyParams.data.username,
+			statusMessage: this.bodyParams.data.statusMessage,
```
Comment (geekgonecrazy) [inline app/api/server/v1/users.js]: would be really cool if like setAvatar this was a rest call.  Imagining the ability to set status from music players, and many other much more practical integrations.

Comment (wreiske) [reply app/api/server/v1/users.js]: Like my smart toaster! "toast is done".

I'll get this coded up when I wake up unless someone else wants to give it a whack before then.

Comment (wreiske) [reply app/api/server/v1/users.js]: OK, pushed. @geekgonecrazy please take a look at the changes and let me know if I should change it.

**Comment by wreiske** [comment]
set-status-api

**Comment by rodrigok** [comment]
@wreiske the tests are failing, can you check?

**Comment by wreiske** [comment]
> @wreiske the tests are failing, can you check?

Can do. Give me a few...

**Comment by wreiske** [comment]
> @wreiske the tests are failing, can you check?

@rodrigok fixed with 8ecbe83. There is something else failing for channels, but I don't think it's related to this.... I could be mistaken.

**Comment by geekgonecrazy** [comment]
> Edit: Looks like the build is failing because https://github.com/lovell/sharp-libvips/releases/download/v8.7.0/libvips-8.7.0-linux-x64.tar.gz is returning a 502.

looks to be working again.  Must have been having an issue at github

**Comment by wreiske** [comment]
> > Edit: Looks like the build is failing because https://github.com/lovell/sharp-libvips/releases/download/v8.7.0/libvips-8.7.0-linux-x64.tar.gz is returning a 502.
> 
> looks to be working again. Must have been having an issue at github

yeah. tests are passing now for mongo 4-0... looks like they are failing at creating a discussion. Unrelated to my changes i think?

**Comment by wreiske** [comment]
Closed in favor of #13933

**Comment by geekgonecrazy** [comment]
🤗 thanks for helping with this feature!

**Pull request closed without merging**